emcc.py

```diff
@@ -2235,27 +2251,31 @@ def check_memory_setting(setting):
     if settings.GLOBAL_BASE != -1:
       exit_with_error("ASan does not support custom GLOBAL_BASE")
 
-    max_mem = settings.INITIAL_MEMORY
+    user_mem = settings.INITIAL_MEMORY
     if settings.ALLOW_MEMORY_GROWTH:
-      max_mem = settings.MAXIMUM_MEMORY
-
-    shadow_size = max_mem // 8
-    settings.GLOBAL_BASE = shadow_size
-
-    sanitizer_mem = (shadow_size + webassembly.WASM_PAGE_SIZE) & ~webassembly.WASM_PAGE_SIZE
-    # sanitizers do at least 9 page allocs of a single page during startup.
-    sanitizer_mem += webassembly.WASM_PAGE_SIZE * 9
-    # we also allocate at least 11 "regions". Each region is kRegionSize (2 << 20) but
-    # MmapAlignedOrDieOnFatalError adds another 2 << 20 for alignment.
-    sanitizer_mem += (1 << 21) * 11
-    # When running in the threaded mode asan needs to allocate an array of kMaxNumberOfThreads
-    # (1 << 22) pointers.  See compiler-rt/lib/asan/asan_thread.cpp.
-    if settings.USE_PTHREADS:
-      sanitizer_mem += (1 << 22) * 4
+      user_mem = settings.MAXIMUM_MEMORY
```

Comment:
This might be very high, like 4GB. Is that not an issue?

Reply:
Yes, I think that just means you get a large shadow region.. its needs to be big enough up front since the shadow region can't grow.    This seems to work ok for our `asan`  tests which all run with `ALLOW_MEMORY_GROWTH`.

Reply:
Ok, sgtm.